core/src/main/java/hudson/model/Slave.java

```diff
@@ -348,9 +349,14 @@ public FilePath getRootPath() {
     /**
      * Web-bound object used to serve jar files for JNLP.
      */
-    public static final class JnlpJar implements HttpResponse {
+    public static class JnlpJar implements HttpResponse {
```

Comment:
Yeah. But the idea is to tweak the default endpoint. In such case there will be no changes required on the agent side, docker images, documentation, etc.